src/main/java/net/minecraftforge/client/ForgeHooksClient.java

```diff
@@ -762,11 +775,33 @@ public static void onInputUpdate(EntityPlayer player, MovementInput movementInpu
     {
         MinecraftForge.EVENT_BUS.post(new InputUpdateEvent(player, movementInput));
     }
-    
+
     public static String getHorseArmorTexture(EntityHorse horse, ItemStack armorStack)
     {
         String texture = armorStack.getItem().getHorseArmorTexture(horse, armorStack);
         if(texture == null) texture = horse.getHorseArmorType().getTextureName();
         return texture;
     }
+
+    public static boolean shouldUseVanillaReloadableListener(IResourceManagerReloadListener listener)
```

Comment:
We need some form of config option to disable this selective reload because you KNOW there will be mods that rely on everything reloading in edge cases. Soi having a opt-out may be feasible.